### PR TITLE
New Sample: Query with CQL filters

### PR DESCRIPTION
**Pull request by shubham7109**
PR to add a new sample to show query data from an OGC API feature service using CQL filters!

### CODE REVIEW

File: kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/query_with_cql_filters/MainActivity.kt
Comment (TADraeseke) [inline kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/query_with_cql_filters/MainActivity.kt:233]: Same with these

File: kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/query_with_cql_filters/MainActivity.kt
Comment (TADraeseke) [inline kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/query_with_cql_filters/MainActivity.kt:170]:
```suggestion
     * Handles CQL Filters in a Bottom Sheet View.
```

File: kotlin/query-with-cql-filters/src/main/res/layout/cql_filters_layout.xml
Comment (TADraeseke) [inline kotlin/query-with-cql-filters/src/main/res/layout/cql_filters_layout.xml:203]: Few places with extra white space in here. Can this file be formatted?

File: kotlin/query-with-cql-filters/src/main/res/values/colors.xml
Comment (TADraeseke) [inline kotlin/query-with-cql-filters/src/main/res/values/colors.xml:14]: delete whitespace

**Review by TADraeseke — CHANGES_REQUESTED** [review]
Really nice looking sample @shubham7109 -- Just some minor suggestions

**Comment by Rachael-E** [comment]
hey @shubham7109 I'll hold off on this review until you've addressed Trevor's comments above, just ping me when you're ready!

File: kotlin/query-with-cql-filters/README.metadata.json
Comment (Rachael-E) [inline kotlin/query-with-cql-filters/README.metadata.json:2]: do Android have categories? check with @TADraeseke what best practice is here. The category should be OGC.

File: kotlin/query-with-cql-filters/README.md
Comment (Rachael-E) [inline kotlin/query-with-cql-filters/README.md:5]: could the preview image show the UI?

File: kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt
Comment (Rachael-E) [inline kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt:1]: this is a new sample, so the copyright should reflect the year the sample was created. In samples where we are updating sample code, we leave the copyright year as is. 

```suggestion
/* Copyright 2021 Esri
```

File: kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt
Comment (Rachael-E) [inline kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt:69]: I read this as the default value the API sets is 1000, which isn't the case (I think Tanner said it was 10!): we could go with the following comment which is used in the design for the other OGC API samples and provides some helpful context to the user:

      // set a limit of 1000 on the number of returned features per request, the default on some services could be as low as 10

File: kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt
Comment (Rachael-E) [inline kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt:73]: is there a return boolean value within any of the Android components you could wire this into it instead of setting a variable? for example, if you have a check box, you could hook it into there. 

a better name for this boolean would be `isDateFilterSelected`: we have boolean naming standards in the Java/Android team, I'll find this document for you for reference!

File: kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt
Comment (Rachael-E) [inline kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt:94]: could remove extra blank line here

File: kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt
Comment (Rachael-E) [inline kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt:100]: could use a plain english comment style here

```suggestion
        // Create a map with a topographic basemap
```

File: kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt
Comment (Rachael-E) [inline kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt:148]: ah I see this comment here. you might want to change the initiation of this value to 3000 then

File: kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt
Comment (Rachael-E) [inline kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt:240]: today I learned Android uses index 0 to refer to the first month of the year!

File: kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt
Comment (Rachael-E) [inline kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt:255]: perhaps you could just instantiate the boolean here rather than at the top of the code: keep the reference to it at the start but instantiate it here?

File: kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt
Comment (Rachael-E) [inline kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt:287]: hook this into the component: it might include an extra line of code to set the check box ticked to false.

File: kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt
Comment (Rachael-E) [inline kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt:286]: this should be 3000 again?

File: kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt
Comment (Rachael-E) [inline kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt:289]: ah now this is confusing me. I thought the month was zero indexed, as the code above shows!

**Review by Rachael-E — APPROVED** [review]
thanks @shubham7109! I've left a few general comments/suggestions for you to consider, nothing major. I've just taken a very general eye over this sample as I'm not an Android/Kotlin developer so I can't comment on the layout/Android specific code.

File: kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt
Comment (TADraeseke) [inline kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt:243]: good opportunity for a scoping function (apply block)

File: kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt
Comment (TADraeseke) [inline kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt:309]: apply block could be good here too

**Review by TADraeseke — CHANGES_REQUESTED** [review]
Pretty slick sample @shubham7109!

Just a couple comments below.

In using the UI just a couple thoughts:

- If the user selects Date Filter on and hits apply, when hitting the FAB again it's surprising that the Date Filter is off. Same with the "From date" Can these be reasonably persisted?

File: kotlin/query-with-cql-filters/README.md
Comment (shubham7109) [reply kotlin/query-with-cql-filters/README.md:5]: It would be challenging to show the UI on landscape mode, since it will cover the `MapView`. Most screenshot's on Android show the map which is why I used it as the preview image.

File: kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt
Comment (shubham7109) [reply kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt:69]: That's a good point, I've added it in!

File: kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt
Comment (shubham7109) [reply kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt:73]: Good point, I have changed the name of to `isDateFilterSelected` and removed the instance variable approach.

File: kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt
Comment (shubham7109) [reply kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt:240]: I didn't know about this either!

File: kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt
Comment (shubham7109) [reply kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt:289]: Good catch, that's fixed.

**Comment by shubham7109** [comment]
Thank you for the comments, I have gone ahead and made the changes. 
@TADraeseke Regarding the date filter. The bottom sheet view resets all the values once user clicks `Apply`, including the date filer to their defaults. I have set it up this way, as iOS uses the same approach as we use similar UI design. I can make the date filter persist, but it will make sense if **all** the UI elements persists as well. Let me know which approach would be preferred!

File: kotlin/query-with-cql-filters/README.metadata.json
Comment (TADraeseke) [inline kotlin/query-with-cql-filters/README.metadata.json:32]:
```suggestion
        "src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt"
```

Can we check the folder path too to make sure it hasn't got dashes

File: kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt
Comment (TADraeseke) [inline kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt:325]: Do we need to capture feature.geometry here?

**Review by TADraeseke — APPROVED** [review]
Thanks @shubham7109 - Let's not worry about preserving the query state if not part of the design.

Can you just double check the file paths below?

After that, good to merge. Thanks!

File: kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt
Comment (shubham7109) [reply kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt:325]: My apologies, did not notice this until after it was merged to `v.next`. I have created #909 to address this.

**Pull request merged**
